### PR TITLE
Diagnostic data for supporting users.

### PR DESCRIPTION
**Pull request by roeierez**

### CODE REVIEW

File: libs/sdk-core/src/breez_services.rs [hunk 1]
```diff
@@ -1850,6 +1858,32 @@ impl BreezServices {
...
+        let payments = serde_json::to_string(
+            &self
+                .persister
+                .list_payments(ListPaymentsRequest::default())?,
+        )?;
+        let channels = serde_json::to_string(&self.persister.list_channels()?)?;
+        let settings = serde_json::to_string(&self.persister.list_settings()?)?;
+        let reverse_swaps = serde_json::to_string(&self.persister.list_reverse_swaps()?)?;
+        let swaps = serde_json::to_string(&self.persister.list_swaps()?)?;
+        let list_id = serde_json::to_string(&self.persister.get_lsp_id()?)?;
```
Comment (dangeross) [inline libs/sdk-core/src/breez_services.rs:1873]:
```suggestion
        let lsp_id = serde_json::to_string(&self.persister.get_lsp_id()?)?;
```

File: libs/sdk-core/src/breez_services.rs [hunk 2]
```diff
@@ -1042,6 +1042,14 @@ impl BreezServices {
         Ok(self.node_api.execute_command(command).await?)
     }
 
+    // Collects various user data from the node and the sdk storage.
+    // This is used for debugging and support purposes only.
+    pub async fn collect_user_data(&self) -> SdkResult<String> {
```
Comment (dangeross) [inline libs/sdk-core/src/breez_services.rs:1047]: I would avoid calling it `collect_user_data`, especially after reading the Apple privacy manifest guide. Maybe something like `get_diagnostic_data`

Comment (roeierez) [reply libs/sdk-core/src/breez_services.rs]: Done

**Comment by erdemyerebasmaz** [comment]
> Does `sdk-flutter/lib/breez_sdk.dart` need updating also?

Yes, it does.

**Comment by roeierez** [comment]
> Does `sdk-flutter/lib/breez_sdk.dart` need updating also?

Added

**Review by dangeross — APPROVED** [review]
LGTM

**Pull request merged**